lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -172,18 +172,15 @@ class HtmlImage implements ui.Image {
 
   @override
   Future<ByteData?> toByteData({ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    if (format == ui.ImageByteFormat.rawRgba) {
+    // TODO(ColdPaleLight): https://github.com/flutter/flutter/issues/89128
+    if (format == ui.ImageByteFormat.rawRgba || format == ui.ImageByteFormat.rawStraightRgba) {
```

Comment:
nit: Please write this as a `switch` statement, and update the comment to make it clear that we're always returning straight rather than premul. I thought we also had an issue to look at using webGL to do this?

Reply:
Done